app/resources/locale/en/en.po

```diff
@@ -1246,90 +1235,108 @@ msgstr "Wrong block format: {{type}}"
+"export this module as new file, unlock the module and \"save as\""
 
-#: app/scripts/services/project.js:97
+#: app/scripts/services/project.js:103
 msgid "You can load it as it is or convert it for the {{name}} board."
 msgstr "You can load it as it is or convert it for the {{name}} board."
 
-#: app/scripts/controllers/menu.js:1328
+#: app/scripts/controllers/menu.js:1347
 msgid ""
-"You can only build at top-level design. Inside submodules you only can "
+"You can only build at the top-level design. Inside submodules, you can only "
 "<strong>Verify</strong>"
 msgstr ""
-"You can only build at top-level design. Inside submodules you only can "
+"You can only build at the top-level design. Inside submodules, you can only "
 "<strong>Verify</strong>"
 
-#: app/scripts/controllers/menu.js:1355
+#: app/scripts/controllers/menu.js:1372
 msgid ""
-"You can only upload  your design at top-level design. Inside submodules you "
-"only can <strong>Verify</strong>"
+"You can only upload at the top-level design. Inside submodules, you can only "
+"<strong>Verify</strong>"
 msgstr ""
-"You can only upload your design at top-level design. Inside submodules you "
-"only can <strong>Verify</strong>"
+"You can only upload at the top-level design. Inside submodules, you can only "
+"<strong>Verify</strong>"
 
 #: app/views/version.html:23
 msgid "You can read the release notes on the"
-msgstr ""
+msgstr "You can read the release notes on the"
 
-#: app/scripts/controllers/menu.js:574
+#: app/scripts/controllers/menu.js:590
 msgid "Your changes will be lost if you don’t save them"
 msgstr "Your changes will be lost if you do not save them"
 
 #: app/views/design.html:25
 msgid "back"
 msgstr "back"
 
-#: app/scripts/services/drivers.js:412 app/scripts/services/utils.js:1122
+#: app/scripts/services/drivers.js:439 app/scripts/services/utils.js:1117
 msgid "{{app}} is required."
 msgstr "{{app}} is required."
 
-#: app/scripts/controllers/menu.js:1099
+#: app/scripts/controllers/menu.js:1119
 msgid "{{board}} datasheet not defined"
 msgstr "{{board}} datasheet not defined"
 
-#: app/scripts/controllers/menu.js:1085
+#: app/scripts/controllers/menu.js:1105
 msgid "{{board}} pinout not defined"
 msgstr "{{board}} pinout not defined"
 
-#: app/scripts/controllers/menu.js:1125
+#: app/scripts/controllers/menu.js:1145
 msgid "{{board}} rules not defined"
 msgstr "{{board}} rules not defined"
 
-#: app/scripts/controllers/menu.js:483 app/scripts/controllers/menu.js:516
+#: app/scripts/controllers/menu.js:499 app/scripts/controllers/menu.js:532
 msgid "{{name}} exported"
 msgstr "{{name}} exported"
 
+#~ msgid "Read only"
+#~ msgstr "Read only"
+
+#~ msgid ""
+#~ "You can only upload  your design at top-level design. Inside submodules "
+#~ "you only can <strong>Verify</strong>"
+#~ msgstr ""
+#~ "You can only upload your design at top-level design. Inside submodules "
+#~ "you only can <strong>Verify</strong>"
+
 #~ msgid ""
```

Comment:
At line 1392, suggestion:

"Because Icestudio is in active development, you will probably want to keep your "
#~ "Icestudio updated to the latest version."

Reply:
Those messages are commented out: they are not used in any part of the code now...